src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -5,207 +5,152 @@
     class="k-code-block"
     :class="[`theme-${theme}`]"
     data-testid="k-code-block"
-    tabindex="0"
   >
     <div
-      v-if="isSearchable && !isSingleLine"
-      class="k-code-block-actions"
+      v-if="showCodeBlockActions"
+      class="code-block-actions"
     >
-      <p
-        class="k-code-block-search-results"
-        :class="{
-          'k-code-block-search-results-has-query': query !== '',
-        }"
+      <KInput
+        v-model="searchQuery"
+        aria-label="Search"
+        class="code-block-search-input"
+        data-testid="code-block-search-input"
+        :error="regExpError !== null"
+        :error-message="regExpError !== null ? regExpError.message : undefined"
+        placeholder="Search..."
+        @input="handleSearch"
       >
-        <template v-if="query === '' && matchingLineNumbers.length === 0">
-          &nbsp;
-        </template>
-
-        <template v-else-if="matchingLineNumbers.length === 0">
-          No results
-        </template>
-
-        <template v-else-if="typeof currentLineIndex === 'number' && !isShowingFilteredCode">
-          {{ currentLineIndex + 1 }} of {{ matchingLineNumbers.length }}
-        </template>
-
-        <template v-else>
-          {{ matchingLineNumbers.length }} {{ matchingLineNumbers.length === 1 ? 'result' : 'results' }}
+        <template #before>
+          <Transition
+            mode="out-in"
+            name="kongponents-fade-transition"
+          >
+            <CloseIcon
+              v-if="searchQuery"
+              aria-label="Clear query"
+              class="clear-query-button"
+              data-testid="clear-query-button"
+              role="button"
+              tabindex="0"
+              title="Clear query"
+              @click="clearQuery"
+              @keydown.enter="clearQuery"
+              @keydown.space="clearQuery"
+            />
```

Comment:
**Issue**: This must use an actual button. There are only disadvantages to "re-implementing" buttons because one has to add a bunch of functionality that are already standard features of `button`: things like appropriate keyboard interactions (here added via two `keydown` listeners), accessibility properties (some added here via `role` and `tabindex`), etc. all come with `button` by default.

```suggestion
            <button
              v-if="searchQuery"
              aria-label="Clear query"
              class="clear-query-button"
              data-testid="clear-query-button"
              type="button"
              @click="clearQuery"
            >
              <CloseIcon title="Clear query" />
            </button>
```

(This will need adjustments in the selectors attached to `.clear-query-button` I believe)

Reply:
Good idea, applied